### ISSUE TITLE
Notify when there was no commited changes

### ISSUE THREAD

**Issue by piranna** [stefanzweifel/git-auto-commit-action#46]
At https://github.com/stefanzweifel/git-auto-commit-action/blob/c887fa4a076abde01ab6c4ca3e970012ae24833f/entrypoint.sh#L20, if there's no content to commit, the action exit sucessfully, but there's no way to know if there were commited changes or repo is untouched. I need to know this to prevent to publish a package to `npm` with nightly changes if there has been no changes. Ideally Github Actions should provide a `neutral` output as it did in legacy version, but it got removed. Maybe one alternative would be to set an environment variable to notify this.

**Comment by stefanzweifel** [comment]
I think there are a couple of things we could do here:

- Add a new input option which will indicate that the Action should fail if no changes were committed (through the exit code).
- Push a value to the Actions output. Your workflow could then interact with the value and stop the publishing process.

I personally like both approaches.

**Comment by piranna** [comment]
I was thinking about both ideas, and I like both of them, too. Problem with the exit code is that you would not be able to diferenciate when it failed or it was a neutral exit, probably would be the less bad of the cases.

What do you mean about "push a value to the Actions output"? Some trick like Azure one to set environment variables between steps, where they are echoed to stdout with a special prefix? Do you have documentation about that?

**Comment by stefanzweifel** [comment]
> What do you mean about "push a value to the Actions output"?

I thought about adding this: https://help.github.com/en/actions/building-actions/metadata-syntax-for-github-actions#outputs

We could do something like this

```shell
echo \"::set-output name=changed_detected::true\"
echo \"::set-output name=changed_detected::false\"
```

You would then have access to the value in your workflow file through `steps.<step_id>.outputs.tag`.

---

As an example, this repository has a workflow, which automatically creates a new release, when a tag is pushed.
The workflow uses the output of one Action or step in another step.

https://github.com/stefanzweifel/git-auto-commit-action/blob/c887fa4a076abde01ab6c4ca3e970012ae24833f/.github/workflows/continuous-deployment.yml#L16-L29

**Comment by piranna** [comment]
> > What do you mean about "push a value to the Actions output"?
> 
> I thought about adding this: [help.github.com/en/actions/building-actions/metadata-syntax-for-github-actions#outputs](https://help.github.com/en/actions/building-actions/metadata-syntax-for-github-actions#outputs)
> 
> We could do something like this
> 
> ```shell
> echo \"::set-output name=changed_detected::true\"
> echo \"::set-output name=changed_detected::false\"
> ```
> 
> You would then have access to the value in your workflow file through `steps.<step_id>.outputs.tag`.

It's almost equal to how [Azure does it](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=azure-devops&tabs=yaml%2Cbatch#set-a-job-scoped-variable-from-a-script) :-) This fine for me :-)

**Comment by stefanzweifel** [comment]
@piranna Just a head up: I've prepared something in #49 and merged it into `master`. I'm probably gonna release a proper version in the upcoming days, as I would like to add #47 to `4.1.0`as well.

You will then be able to use the output like this:
https://github.com/stefanzweifel/git-auto-commit-action/blob/956a47433ba7e7968b6b72ee72f9e4ca4e8ca2a2/.github/workflows/git-auto-commit.yml

Will close this issue when I've tagged a new version.

**Comment by piranna** [comment]
> I'm probably gonna release a proper version in the upcoming days, as I would like to add #47 to `4.1.0`as well

Great, that would be useful to me too :-) I'm willing to use that feature at nightly pipelines of https://github.com/projectlint/OS-lifecycle :-)

**Comment by stefanzweifel** [comment]
Hello again,

I've just released `v4.1.0` which has the changes made in #49 in it.

https://github.com/stefanzweifel/git-auto-commit-action/releases/tag/v4.1.0

Hope you can utilise this in your workflows.

**Comment by piranna** [comment]
> Hope you can utilise this in your workflows.

Nice, now it's working :-)

https://github.com/projectlint/OS-lifecycle/runs/490127931?check_suite_focus=true

Thank you so much! :-D
